src/indicators/drought/arc2_precipitation.py

```diff
@@ -0,0 +1,884 @@
+            )
+            .reset_index(drop=True)
+            .assign(
+                adm_col=lambda x: x.adm_col.str[0],
+                ds_first_date=lambda x: x.ds_confirmation
+                - pd.to_timedelta(self.rolling_window - 1, unit="d"),
+                ds_duration=lambda x: (
+                    x.ds_last_date - x.ds_first_date
+                ).dt.days
+                + 1,
+            )
+            .rename(columns={"adm_col": adm_col})
+        )
+
+        # Calculate rainfall within each dry spell
+        precip_df = self.load_aggregated_data()
+        precip_col = precip_df.columns[1]
+
+        total_df = pd.merge(
+            left=ds_df, right=precip_df, on=adm_col, how="outer"
+        )
+
+        # Filter dates to just those within the dry spell and sum
+        total_df = total_df[total_df[t_col] >= total_df["ds_first_date"]]
+        total_df = total_df[total_df[t_col] <= total_df["ds_last_date"]]
+        total_df = (
+            total_df.groupby([adm_col, "ds_confirmation"])[precip_col]
+            .agg("sum")
+            .reset_index()
+        )
+
+        # Add dry spell rainfall to main data frame
+        total_df.rename(columns={precip_col: "ds_rainfall"}, inplace=True)
+
+        ds_df = pd.merge(
+            left=ds_df,
+            right=total_df,
+            on=[adm_col, "ds_confirmation"],
+            how="left",
+        )
+
+        # Re-arrange dry spell data frame
+        cols = [
+            adm_col,
+            "ds_first_date",
+            "ds_confirmation",
+            "ds_last_date",
+            "ds_duration",
+            "ds_rainfall",
+        ]
+        ds_df = ds_df[cols]
+
+        ds_fp = self._get_dry_spell_filepath()
+        ds_df.to_csv(ds_fp, index=False)
+
+        return ds_df
+
+    def _get_dry_spell_filepath(self) -> Path:
+        """
+        Return filepath to ARC2 rolling sum values.
+        """
+        directory = self._get_directory(PROCESSED_DATA_DIR)
+        filename = (
+            f"arc2_{self.agg_method}_{self.country_iso3}_"
+            f"dry_spells_{self.rolling_window}_days_"
+            f"{self.range_x[0]}_{self.range_x[1]}_"
+            f"{self.range_y[0]}_{self.range_y[1]}_"
+            f"main.csv"
+        )
+        return directory / filename
+
+    def load_dry_spell_data(self, filter: bool = True) -> pd.DataFrame:
+        """
+        Load dry spells classified through method.
+        """
+        fp = self._get_dry_spell_filepath()
+        df = pd.read_csv(
+            filepath_or_buffer=fp,
+            parse_dates=["ds_first_date", "ds_confirmation", "ds_last_date"],
+        )
+        if filter:
+            t_col = "ds_confirmation"
+            df = df[
+                (df[t_col].dt.date >= self.date_min)
+                & (df[t_col].dt.date <= self.date_max)
+            ]
+        return df
+
+    def _write_to_monitoring_file(self, dry_spells: Union[None, int] = None):
```

Comment:
this function seems unused and `_get_monitoring_filepath` undefined? Fine for now, just add acomment